### DISCUSSION TITLE
Usage of different web server or disabling certificate fetch/check

### DISCUSSION THREAD

**Discussion by SasuGamer** [outline/outline#7343] [Self Hosting] [answered]
Hey there. While all other services just run fine I get an error from https-portal which is unable to get a certificate for my domain.

I wanted to ask: is there any .env variable or other option to use https-portal without it needing a certificate?
I have a certificate myself which I will use but can't for now as I need to move stuff from another server (which has the certificate) to this which I can only do if my instance of Outline is running.

I am not sure why a web server needs a certificate to just run? I understand that for security reasons I need a certificate and I will use one later in time. Also for another instance I want to install I need to turn this check off because I am in an airgapped environment with my own certificates.

Thanks in advance!

**Comment by tommoor** [comment]
>  is there any .env variable or other option to use https-portal without it needing a certificate?

The point of `https-portal` is to auto-provision a certificate, if you don't want one then it makes sense to just remove it from the deployment entirely.

> I am not sure why a web server needs a certificate to just run?

Outline can run without HTTPS, make sure the `FORCE_HTTPS` env is set to false.

You can have Outline serve your own certificate in recent versions – see the documentation here:
https://docs.getoutline.com/s/hosting/doc/ssl-pzk7WO8d1n

**Reply by SasuGamer** [reply]
> > is there any .env variable or other option to use https-portal without it needing a certificate?
> 
> The point of `https-portal` is to auto-provision a certificate, if you don't want one then it makes sense to just remove it from the deployment entirely.
> 
> > I am not sure why a web server needs a certificate to just run?
> 
> Outline can run without HTTPS, make sure the `FORCE_HTTPS` env is set to false.
> 
> You can have Outline serve your own certificate in recent versions – see the documentation here: https://docs.getoutline.com/s/hosting/doc/ssl-pzk7WO8d1n

I now tried to get it running without HTTPS-PORTAL. Took a few troubleshooting steps but it starts up. But now I have the problem that connected auth providers tell me the auth URL is invalid. Tried it with Discord and OAUTH via authentik. Does this have to do anything with HTTPS-PORTAL? Does Outline auth need a certificate to redirect correctly? 

Thanks in advance!

**Reply by tommoor** [reply]
> auth providers tell me the auth URL is invalid

What is the error message?

> Does Outline auth need a certificate to redirect correctly?

No, it's not required.

**Reply by SasuGamer** [reply, marked as answer]
@tommoor Hi there thanks for your quick answer. It just seems that my incompetence and too long nights on this hindered me from a solution. For the authentik-instance me and my colleagues checked which way the communication goes and it seems that from our infrastructure there were two ways. One directly to the load balancer and one via a proxy. We set it to only use the proxy and it just works fine. For the Discord problem I saw that (whyever) the auth redirect URL was redirecting to http. I set https in all configs and my NGINX but for no reason it still wants to use http. I changed my config to use HTTP as well and now it works. While this is not intended for long time use it is working for now.
